devito/dimension.py

```diff
@@ -31,6 +30,11 @@ def __new__(cls, name, **kwargs):
 
     def __str__(self):
         return self.name
+    
+    @property
+    def dtype(self):
+        #TODO: Do dimensions really need a dtype?
```

Comment:
Something in the DLE breaks without it. See https://github.com/opesci/devito/pull/453/commits/0db90be38fd88adb303f8a8a7ff067b7e8eb3362

Reply:
@mlange05 do you think it's feasible to have ``Dimension``s inherit from ``devito.Symbol``, rather then ``sympy.Symbol``? Our symbols carry a dtype already. I'm asking because I seem to remember you were the one playing with this stuff not so long ago?